src/components/engagement/engagement.service.ts

```diff
@@ -211,7 +208,10 @@ export class EngagementService {
 
   private async verifyCreateEngagement(projectId: ID, session: Session) {
     const project = await this.resources.load(IProject, projectId);
-    const projectPrivileges = this.privileges.for(session, IProject, project);
+    const projectPrivileges = this.privileges.for(session, IProject, {
+      ...project,
+      project,
+    } as any);
```

Comment:
I'm curious why the awkward spreading here.

Reply:
The project object is used as the context object when checking "can create engagement".
It works because project has the same attributes as engagement: sensitivity, isMember, etc.
But the condition I just added is looking for `context.project`, so we also need project at a sub key here.

What would be better altogether is to check the create permission after creating it.
At that point we'd have the object like normal. And then we can just rollback the transaction if permissions fail.
Also EdgeDB will be checking this too with APs...so these statements could also just go away after migration.

Reply:
That makes sense!

I'm ok with this, especially if we can move away from it. Normally I'd want to provide what specific fields are needed or trim down a copy of the `project`, but it's ok for now.

Reply:
It's hard to know what's needed is the problem. The policies are very flexible. I believe it could be possible to add a build step to generate the context types for each resource, based on what's used in policies.